src/pages/challenges/get-healthy.js

```diff
@@ -1,34 +1,57 @@
-import Image from 'next/image'
+import styles from '../../styles/Challenges.module.css'
 import HeaderCard from 'components/cards/HeaderCard'
 import ChallengeCategoryList from 'components/challenges/ChallengeCategoryList'
-import styles from '../../styles/Challenges.module.css'
 import DatePrinter from '../../components/layout/DatePrinter'
+import {
+	getTodayTasksByCategory,
+	getTodayScoreByCategory
+} from 'model/helpers/today-tasks'
+import { useState } from 'react'
+import { useUserContext } from 'context/UserContext.js'
+
+export default function GetHealthy({ tasks, progress }) {
+	const [data, setData] = useState({ tasks: tasks, progress: progress })
+	const userId = 1
+	const ctx = useUserContext()
 
-export default function GetHealthy({ tasks }) {
 	const completedHandler = async (taskId) => {
-		const response = await fetch('http//localhost:3000/api/update-status', {
+		const taskPoints = data.tasks.find(
+			(task) => task.id === Number(taskId)
+		).points
+
+		const response = await fetch('http://localhost:3000/api/update-status', {
 			method: 'PUT',
 			headers: {
 				'Content-Type': 'application/json'
 			},
-			body: JSON.stringify({ taskId: 1, date: '2023-03-29' })
+			body: JSON.stringify({ userId: userId, taskId: taskId })
 		})
 
 		if (response.ok) {
-			console.log(`task ${taskId} has been ticked`)
+			setData((prevState) => {
+				const updatedTasks = prevState.tasks.map((task) =>
+					task.id == taskId ? { ...task, status: 1 } : task
+				)
+				const completedTasks = updatedTasks.filter(
+					(task) => task.status === 1
+				).length
+				const totalTasks = updatedTasks.length
+				const newProgress = Math.round((completedTasks / totalTasks) * 100)
```

Comment:
A thought - do you need `progress` to live in the `useState`? It's a number that's calculated from the `tasks` state, so you would also be fine to get the progress in the render (ie outside of any hooks).